sim/src/cap.rs

```diff
@@ -99,6 +74,33 @@ impl CapSimState {
         true
     }
 
+    pub fn validate_path(
+        &mut self,
+        req: &PathRequest,
+        path: Path,
+        now: Time,
+        car: CarID,
+        map: &Map,
+    ) -> Option<Path> {
+        if self.allow_trip(now, car, &path) {
+            return Some(path);
+        }
+
+        // TODO Make the responses configurable: abort the trip, reroute, delay an hour, switch
+        // modes. Where should this policy be specified? Is it simulation-wide?
+
```

Comment:
Building `avoid_lanes` and/or the dijkstra is SLOW. Extremely noticeable in debug mode. Will think about optimizations after it's a bit more clear what direction this project takes. Some optimizations could include rebuilding the contraction hierarchies every time a zone is filled/becomes available again. That seems intense, but it should be num_zones * 24 at most.

Reply:
I could imagine a "live edit" scenario which might be further reason to make this run fast one day.